Sources/FoundationInternationalization/URLParser+ICU.swift

```diff
@@ -15,7 +15,14 @@ import FoundationEssentials
 
 internal import _FoundationICU
 
-internal final class UIDNAHookICU: UIDNAHook {
+#if !FOUNDATION_FRAMEWORK
+@_dynamicReplacement(for: _uidnaHook())
+private func _uidnaHook_localized() -> UIDNAHook.Type? {
+    return UIDNAHookICU.self
+}
+#endif
+
+private struct UIDNAHookICU: UIDNAHook {
```

Comment:
Did I misread it or was `UIDNAHookICU` changed from `final class` to a `struct` 👀?  Is this intentional?

Reply:
Yeah, the only reason it was a `final class` was so that `_typeByName` would work (it can only lookup classes). I changed it to a `struct` to be more in-line with how we typically write swift types now that a class isn't required

Reply:
Thanks that makes total sense.